### PR TITLE
adds s3 notification event and scheduled event

### PR DESCRIPTION
**Pull request by sdelamo**
When you work with Lambda. You define the trigger your Lambda Function reacts to. AWS Provides a library (AWS Lambda Events) with some Java POJOs to represent those triggers. 

This PR gives users the ability to generate a Micronaut Functions which is triggered by some of the [AWS Lambda Events](https://mvnrepository.com/artifact/com.amazonaws/aws-lambda-java-events). 

We already have some guides which use some triggers beyond HTTP triggers:

- [Micronaut Lambda Scheduled](https://guides.micronaut.io/latest/micronaut-scheduled.html)
- [S3 Notification Event](https://guides.micronaut.io/latest/micronaut-aws-lambda-s3-event.html)

### CODE REVIEW

File: starter-core/src/main/java/io/micronaut/starter/feature/aws/AmazonApiGateway.java [hunk 1]
```diff
@@ -18,12 +18,13 @@
...
 import io.micronaut.starter.application.ApplicationType;
 import io.micronaut.starter.feature.Category;
+import io.micronaut.starter.feature.function.awslambda.AwsLambda;
 import jakarta.inject.Singleton;
 
 @Singleton
-public class AmazonApiGateway extends CdkFeature {
-    public AmazonApiGateway(Cdk cdk) {
-        super(cdk);
+public class AmazonApiGateway extends AwsLambdaRelatedFeature implements AwsApiFeature {
```
Comment (timyates) [inline starter-core/src/main/java/io/micronaut/starter/feature/aws/AmazonApiGateway.java:25]: Is the description here still correct?  It mentions the CDK, but the CDK has been removed from here?

File: test-aws/src/test/groovy/io/micronaut/starter/core/test/aws/CreateAwsLambdaScheduledEventSpec.groovy [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    }
+
+    @Unroll
+    void 'create-#applicationType with features aws-lambda-scheduled-event #lang and #build and test framework: #testFramework'(ApplicationType applicationType,
+                                                                                                                             Language lang,
+                                                                                                                             BuildTool build,
+                                                                                                                             TestFramework testFramework) {
+        given:
+        List<String> features = ['aws-lambda-scheduled-event']
+        generateProject(lang, build, features, applicationType, testFramework)
```
Comment (timyates) [inline test-aws/src/test/groovy/io/micronaut/starter/core/test/aws/CreateAwsLambdaScheduledEventSpec.groovy:25]: The generated project here has no tests.  Should it have?

File: test-aws/src/test/groovy/io/micronaut/starter/core/test/aws/CreateAwsLambdaS3NotificationEventSpec.groovy [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    void 'create-#applicationType with features aws-lambda-s3-event-notification #lang and #build and test framework: #testFramework'(ApplicationType applicationType,
+                                                                                                                             Language lang,
+                                                                                                                             BuildTool build,
+                                                                                                                             TestFramework testFramework) {
+        given:
+        List<String> features = ['aws-lambda-s3-event-notification']
+        generateProject(lang, build, features, applicationType, testFramework)
+
+        when:
+        String output = executeBuild(build, "test")
```
Comment (timyates) [inline test-aws/src/test/groovy/io/micronaut/starter/core/test/aws/CreateAwsLambdaS3NotificationEventSpec.groovy:28]: The generated project here has no tests. Should it have?

**Review by timyates — CHANGES_REQUESTED** [review]
Requesting changes for feedback

File: test-aws/src/test/groovy/io/micronaut/starter/core/test/aws/CreateAwsLambdaScheduledEventSpec.groovy (hunk 2 above)
Comment (sdelamo) [reply test-aws/src/test/groovy/io/micronaut/starter/core/test/aws/CreateAwsLambdaScheduledEventSpec.groovy:25]: the handler returns `Void`. Thus, it is intentional the lack of generated test.

File: test-aws/src/test/groovy/io/micronaut/starter/core/test/aws/CreateAwsLambdaS3NotificationEventSpec.groovy (hunk 3 above)
Comment (sdelamo) [reply test-aws/src/test/groovy/io/micronaut/starter/core/test/aws/CreateAwsLambdaS3NotificationEventSpec.groovy:28]: the handler returns `Void`. Thus, it is intentional the lack of generated test.

File: starter-core/src/main/java/io/micronaut/starter/feature/aws/AmazonApiGateway.java (hunk 1 above)
Comment (sdelamo) [reply starter-core/src/main/java/io/micronaut/starter/feature/aws/AmazonApiGateway.java:25]: You can use it in combination with CDK but you can use it without it.

**Comment by timyates** [comment]
Tests fail as they are expecting

```
Using BASE_IMAGE: ghcr.io/graalvm/native-image:ol8-java11-22.0.0.2
```

But the logs say

```
Using BASE_IMAGE: ghcr.io/graalvm/native-image:ol8-java11-22.1.0
```

**Comment by timyates** [comment]
Running

```
./gradlew :starter-core:micronautVersionInfo
```

Also seems to give us a mixed set of graal versions in `starter-core/build/version-info/micronaut-versions.properties` 🤔

```
graal.version=22.0.0.2
graalSdk.version=22.1.0.1
```

**Review by timyates — APPROVED** [review]
LGTM apart from the 2 questions above ☝️

**Comment by sdelamo** [comment]
> Running
> 
> ```
> ./gradlew :starter-core:micronautVersionInfo
> ```
> 
> Also seems to give us a mixed set of graal versions in `starter-core/build/version-info/micronaut-versions.properties` 🤔
> 
> ```
> graal.version=22.0.0.2
> graalSdk.version=22.1.0.1
> ```

see: https://github.com/micronaut-projects/micronaut-core/pull/7478

**Comment by timyates** [comment]
Cool, `3.5.0-SNAPSHOT` has been published now, so the `graal.version` is correct 👍 

I'll push a fix for those tests

**Pull request merged**